.github/workflows/main.yaml

```diff
@@ -0,0 +1,312 @@
+name: Build and Deploy Image
+
+on:
+  workflow_dispatch:
+    inputs:
+      tag:
+        description: 'Define the tag for the code'     
+        required: false
+      branch:
+        description: 'Define branch name'     
+        required: false
+        default: 'main'
```

Comment:
These inputs are not being used by the actions. What's the intent with adding these?

Reply:
The inputs when defined will build the image based on tag or branch.

Reply:
So just a matter of fleshing out that implementation?